core/lib/state/src/storage_view.rs

```diff
@@ -52,6 +52,13 @@ pub struct StorageView<S> {
     metrics: StorageViewMetrics,
 }
 
+impl<S> StorageView<S> {
+    /// Returns the modified storage keys
+    pub fn modified_storage_keys(&self) -> &HashMap<StorageKey, StorageValue> {
```

Comment:
There is already a method `modified_storage_keys(&self)` which is a part of `WriteStorage` implementation. Do you need a dedicated one here? If so please explain why. If you need it indeed please reuse it in `WriteStorage` implementation